docs/can-stache-element.md

```diff
@@ -172,40 +172,348 @@ If needed, [can-stache-element/lifecycle-hooks.connected] and [can-stache-elemen
+    <p>I say "<i>{{message}}</i>"!</p>
+  </can-template>
+</hello-world>
+```
+
+Here's what you need to know about `<can-template>`:
+
+- Every `<can-template>` __MUST__ have a name attribute. This is
+  the name of the property on the custom
+  element that will be set to the template. In the previous example, both `<hello-world>`'s will be created
+  with a `messageTemplate` property:
+
+  ```js
+  document.querySelector("hello-world").messageTemplate //-> templateFunction()
+  ```
+
+- You can have multiple `<can-template>`s within a custom element.  For example,
+  the following passes two templates to configure `<hello-world>`:
+
+  ```html
+  <hello-world>
+    <can-template name="greetingTemplate">
+      <b>{{greeting}}</b>
+    </can-template>
+    <can-template name="subjectTemplate">
+      <b>{{subject}}</b>
+    </can-template>
+  </hello-world>
+  ```
+
+- `<can-template>`s have the same scope of the custom element, __plus__
+  a `LetScope` that the custom element can optionally provide. This means
+  that a `{{this.someData}}` immediately outside the `<hello-world>` will
+  reference the same value as `{{this.someData}}` within a `<can-template>`:
+
+  ```html
+  {{this.someData}}
+  <hello-world>
+    <can-template name="messageTemplate">
+      <h1>{{message}}</h1>
+      <p>Also: {{this.someData}}</p>
+    </can-template>
+  </hello-world>
+  ```
+  @highlight 1,5
+
+  The custom element can add additional data, like `message`, to these
+  templates. We will see how to do that in the next section.
+
+#### Calling the templates with `{{this.template()}}`
+
+Once templates are passed to a custom element, you can call those templates
+within the `StacheElement`'s [can-stache-element/static.view]. For example,
+`<hello-world>` might call a passed `messageTemplate` as follows:
+
+```html
+<my-app></my-app>
+
+<script type="module">
+import { StacheElement } from "can/everything";
+
+class HelloWorld extends StacheElement {
+  static view = `
+    <div>{{ this.messageTemplate() }}</div>
+  `;
+  static props = {
+    messageTemplate: {type: Function, required: true},
+    message: "Hello World"
+  }
+}
+customElements.define("hello-world", HelloWorld);
+
+class App extends StacheElement {
+  static view = `
+    <hello-world>
+      <can-template name="messageTemplate">
+        <h1>{{ message }}</h1>
+      </can-template>
+    </hello-world>
+  `;
+}
+customElements.define("my-app", App);
+</script>
+```
+@codepen
+@highlight 8,17-26,only
+
+While the above will render the passed `messageTemplate`, it will not provide it
+a `{{message}}` variable that can be read. You can pass values into a template
+with a [can-stache/expressions/hash]. The following passes the message:
+
+```html
+<my-app></my-app>
+
+<script type="module">
+import { StacheElement } from "can/everything";
+
+class HelloWorld extends StacheElement {
+  static view = `
+    <div>{{ this.messageTemplate( message = this.message) }}</div>
```

Comment:
@cherifGsoul `this.messageTemplate( message = this.message)` -> `this.messageTemplate(message = this.message)`